LocalPackages/DataBrokerProtection/Sources/DataBrokerProtection/Scheduler/DataBrokerProtectionAgentManager.swift

```diff
@@ -177,11 +187,29 @@ extension DataBrokerProtectionAgentManager {
     }
 }
 
+private extension DataBrokerProtectionAgentManager {
+
+    /// Starts either Freemium (scan-only) or Subscription (scan and opt-out) scheduled operations
+    /// - Parameters:
+    ///   - showWebView: Whether to show the web view or not
+    ///   - operationDependencies: Operation dependencies
+    ///   - completion: Completion handler
+    func startFreemiumOrSubscriptionScheduledOperations(showWebView: Bool,
+                                                        operationDependencies: DataBrokerOperationDependencies,
+                                                        completion: ((DataBrokerProtectionAgentErrorCollection?) -> Void)?) {
+        if freemiumPIRUserState.isActiveUser {
```

Comment:
Note: Thinking more here - This makes me uncomfortable because:

* It relies on the `accountManager` that is injected into `FreemiumPIRUserState` being the same instance as the one being used in other parts of the code
* Is the wrong instance was injected, we could end up not running opt-outs for subscribers

So, I plan to refactor this code (removing `AccountManager` from `FreemiumPIRUserState` so that this check uses the same `accountManager` instance used by the background agent stopper etc. Task is [here](https://app.asana.com/0/0/1208197280287450/f). I decided not to make these changes as part of this PR, as the changes here are large enough already. However, I think we can go ahead with this PR first, as it’s to a feature branch and I’ll follow-up with the refactoring.

Reply:
I ended up having the same thought whilst reviewing AgentStopper but kinda from a different angle. The default implementation of FreemiumPIRUserState is quite dynamic (i.e. it's backed by user defaults and could change without that specific instance of the object being modified), but to me the name suggests it would be a static thing (like an enum case).

Because of that, there might be an alternative solution, which would just be to rename it (E.g. FreemiumPIRUserStateManager). It'd obviously still break if different things got different implementations of FreemiumPIRUserStateManager, but then it would be more expected

Reply:
Thanks @THISISDINOSAUR , thats a good suggestion. I’ll make that renaming change here, and still follow-up with some work to refactor the logic a bit as mentioned above.